File: hrms-frontend/src/services/accountRequestService.js
import axiosClient from "./axiosClient";

const accountRequestService = {
  getRequests: (page = 0, size = 10, filters = {}) => {
    const {
      status = "all",
      search,
      departmentId,
      positionId,
      lineId,
    } = filters;

    return axiosClient.get(`/account-requests`, {
      params: {
        status,
        page,
        size,
        // chỉ gửi khi có giá trị
        ...(search ? { search } : {}),
        ...(departmentId ? { departmentId } : {}),
        ...(positionId ? { positionId } : {}),
        ...(lineId ? { lineId } : {}),
      },
    });
  },

  approveRequest: (id) => {
    return axiosClient.post(`/account-requests/${id}/approve`);
  },

  rejectRequest: (id) => {
    return axiosClient.post(`/account-requests/${id}/reject`);
  },
};

export default accountRequestService;

File: hrms-backend/src/main/java/sep490/com/example/hrms_backend/service/AccountRequestService.java
package sep490.com.example.hrms_backend.service;

import org.springframework.data.domain.Page;
import sep490.com.example.hrms_backend.dto.AccountRequestDTO;

public interface AccountRequestService {

    void approveRequest(Long requestId);
    void rejectRequest(Long requestId);
    Page<AccountRequestDTO> getRequestsByStatus(
            String status, int page, int size,
            String search, Long departmentId, Long positionId, Long lineId
    );


}
